src/screens/FeedScreen/hooks/useFeedHeader.js

```diff
@@ -66,17 +66,20 @@ const useFeedHeader = ({actor, source}) => {
     handleNavigate(selfUserId);
   };
 
-  const onBackNormalUser = () => {
+  const onBackNormalUser = (isFromFeeds) => {
     if (viewPostTimeIndex >= 0) id = feeds[viewPostTimeIndex]?.id;
     if (source && id) sendViewTimePost(id);
 
     resetTimer();
-    navigation.navigate('HomeTabs', {
-      screen: 'Feed',
-      params: {
-        isGoBack: true
-      }
-    });
+    if (isFromFeeds) {
+      return navigation.navigate('HomeTabs', {
+        screen: 'Feed',
+        params: {
+          isGoBack: true
+        }
+      });
+    }
+    return navigation.goBack();
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes made to the `onBackNormalUser` function seem to be logically sound. The addition of the `isFromFeeds` parameter allows for conditional navigation, either going back to the previous screen or navigating to the 'Feed' screen based on its value. However, it's important to ensure that this new parameter is always provided when calling this function to avoid unexpected behavior. If there are cases where this function could be called without this parameter, you should consider providing a default value.

```diff
-  const onBackNormalUser = (isFromFeeds) => {
+  const onBackNormalUser = (isFromFeeds = false) => {
```
This change will ensure that if `isFromFeeds` is not provided, it will default to `false`, and the function will behave as it did prior to these changes.


<!-- This is an auto-generated comment by OSS CodeRabbit -->